### ISSUE TITLE
AudioPlayer Read, shadowThreadFunc() crash

### ISSUE THREAD

**Issue by flagoworld** [superpoweredSDK/Low-Latency-Android-iOS-Linux-Windows-tvOS-macOS-Interactive-Audio-Platform#508]
I have gotten many of these crashes. Averaging 30 per month. I've attached a zip containing 5 of the most recent stack traces.

Appears to happen for local files as well as progressive streaming.

My code is not really doing much at the time of this happening. As you can see on the main thread nothing is really happening.

There is no pattern that I have figured out yet... Random files, urls, etc will work most of the time and occasionally crash. A number of them are in low ram conditions, but that I believe is a coincidence. Once of them attached is not, and looking back through previous logs this happens a lot when not low on RAM.

[crashes.zip](https://github.com/superpoweredSDK/Low-Latency-Android-iOS-Linux-Windows-tvOS-macOS-Interactive-Audio-Platform/files/2315416/crashes.zip)

**Comment by superpoweredSDK** [comment]
1453 empty as well... there is nothing in that code area which runs

**Comment by flagoworld** [comment]
I’ve seen this before, where the offset is a bit off. When I’ve seen it, the problem was on a nearby line but something caused it to be a few lines off. Not really sure why was when it happened I was able to use the stack trace to infer which line was the likely problem. Not sure if that’s possible here.

**Comment by superpoweredSDK** [comment]
Yes, but there is really not much around here. I think the line number is totally off.

**Comment by superpoweredSDK** [comment]
Because we are still in the dark, let's wait for the next Superpowered update (coming in a few weeks) and let's see if it's better, because it will have some changes under the hood.

**Comment by superpoweredSDK** [comment]
update is out

**Comment by flagoworld** [comment]
Thanks. I'll install and get back in a few weeks once I have collected some data.

**Comment by eduardhasanaj2** [comment]
@flagoworld 
please can you tell me what server do you use for progressive download?

**Comment by gaborszanto** [comment]
Could you collect some data so far?

**Comment by flagoworld** [comment]
We haven't released the version with the latest superpowered but will be doing so in a few days. Give it about 3 weeks and I'll have some real data for you. Sorry I can't have anything sooner.

**Comment by gaborszanto** [comment]
Any data please?

**Comment by flagoworld** [comment]
Alright, here's the latest crash with the latest versions of superpowered

[crash.txt](https://github.com/superpoweredSDK/Low-Latency-Android-iOS-Linux-Windows-tvOS-macOS-Interactive-Audio-Platform/files/2532480/crash.txt)

**Comment by gaborszanto** [comment]
Unfortunately the line number is still off, it points to no code.
I see that during this latest crash a player is destroyed (or just a new open() happens).
At which point in your code do you destroy a player?
At which point in your code do you open() a player?

**Comment by gaborszanto** [comment]
Yes, theoretically we fixed that, but can you please try it just to see if this is the case?

**Comment by flagoworld** [comment]
I haven't personally experienced this crash in a little while on my devices. However that crash came in from a user using the app. I don't know what they were doing 3 identical crashes from 2 users in the past 2 days.

**Comment by gaborszanto** [comment]
Until we don't know 100% percent that this is the error, we can't fix. You need to be able to reproduce the crash. Until then we are just guessing what can be the problem.

**Comment by flagoworld** [comment]
I believe I found a clue and maybe figured this out. I was developing and typo'd some code, leading me to setPosition() with a negative float as the position. This caused the crash I've been experiencing. I don't see anywhere that this should happen outside of my bug, but this may be a clue to figuring out this crash. I have added a check on my end to make sure the value is positive. We'll see if that fixes it. I'm thinking you guys might want to add the check on your end too so negative input doesn't crash but instead sets the position to 0 or something.
